plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapper.java

```diff
@@ -141,6 +146,12 @@ Use VIR_DOMAIN_XML_SECURE (value = 1) prior to v1.0.0.
             xmlDesc = dm.getXMLDesc(xmlFlag);
             xmlDesc = replaceIpForVNCInDescFile(xmlDesc, target);
 
+            String oldIsoVolumePath = getOldVolumePath(disks, vmName);
+            String newIsoVolumePath = getNewVolumePathIfDatastoreHasChanged(libvirtComputingResource, conn, to);
+            if (newIsoVolumePath != null && !newIsoVolumePath.equals(oldIsoVolumePath)) {
+                s_logger.debug("Editing mount path");
+                xmlDesc = replaceDiskSourceFile(xmlDesc, newIsoVolumePath, vmName);
+            }
```

Comment:
I think even this part could be factorred out. the method is still almost 200 lines. (no biggy)